### ISSUE TITLE
[BUG] When Import nested Component

### ISSUE THREAD

**Issue by ZiggyReQurv** [Baboo7/strapi-plugin-import-export-entries#81]
Hi Baboo7,
we are building a question manager with Strapi and we found a possible bug in the import process.
We are currently using Strapi v4.5.1 and the last version of your plugin (v 1.18.0) and Postgres as DB.
We've created a collection called Questions composed of two-level components, as shown in image1
![Image1](https://user-images.githubusercontent.com/115216906/203075769-776bc56c-c96b-4df8-9c4e-d30cb2aa8656.png)

When we create a new record in this collection, the result is image2

![Image2](https://user-images.githubusercontent.com/115216906/203075803-2eda5e20-7764-442b-b109-03ebdf403ccf.png)

Our issue happened when we tried to export data from one DB and importing the exact same data to another DB (same version): what happen is that the result of the import looks like we have the same content as the previous image but when we make any changes in any record and then save, we have a duplication of the options component, as shown in image3. This does not happen in the question component, I imagine because in not a repeatable component.
![Image3](https://user-images.githubusercontent.com/115216906/203075846-2a4bddb7-4c71-44d5-b74d-3f0f74232c44.png)

We made some digging in the DB and what we discover is that some data during the import process is not properly imported. 
The table shown in the image4 is called questions_components and contains the linking information between the Questions collection and its components.
![Image4](https://user-images.githubusercontent.com/115216906/203075904-2cc9dff1-7c24-46d5-8bd9-c4286a2d39b6.png)

As you can see, the component_type column does not acquire the data of the component
If you make any change in the record and then save (as described in image3), the data is duplicated as shown in image5, this time with the component_type properly set.
![Image5](https://user-images.githubusercontent.com/115216906/203075960-c2bdca19-63ea-429c-9cba-76ea4af2a8b2.png)

After we found this behavior, we tried to import the same data to a new clean DB and immediately went to set the component_type column manually; we basically hardcoded the string we expected to find. The solution of inserting this data manually worked out, that's how we found out about the problem. In image 6 you can see what we hard-coded in the component_type.
![Image6](https://user-images.githubusercontent.com/115216906/203076248-d80dc0c0-7ea3-4377-8ea5-d7defb5322b3.png)

We then tried to make a change and save a record and the data was not duplicated, confirming our thoughts. After this edit, the records of the collection behave as expected.
To be clear, the import DID NOT have any errors, the process went through with no errors. 
That's how we ended up thinking that during the import process the data of the component_type column is not set.
This could be a limitation of the plugin when working with nested components.
Let me know if you need any additional information.
Thank you.

**Comment by mitenka** [comment]
Hey @ZiggyReQurv we faced the same issue and are trying to fix it now. By any chance have you had any success with this issue? Thank you

**Comment by exploevo** [comment]
HI @mitenka the developer never replies back to us so since we are using Postgres as a database we are starting to work directly from the database interface to export end import the date using the command:
pg_dump --no-owner nomedatabase > filedumb.sql

psql databasericevente < filedump.sql

it was a pretty good turnaround that give us very good results and speed up our work.
I hope that this could be useful to you
Best wishes

**Comment by mitenka** [comment]
Hi @exploevo, thank you for sharing your experience! I guess the issue #91 is about the same thing. Thus I've created PR #97 to fix it. Hopefully it works for you as well. Feel free to report in case of any corner cases

**Comment by Kyle772** [comment]
https://discord.com/channels/1021121498276171786/1021121949537153145/1049829680573849661

A user on the discord channel has figured out a work around until this is fixed

Here are the steps:
1) Run an import
2) Query through the db with pgadmin or another db management tool
3) Anything that ends in "_components" is a culprit
4) Look for component_type === null
5) Grab the value from the attributes.<name>.component key in each of your strapi schemas 
6) do a bulk update on all component_types that are null and set it to the value from step 5
6.5) If you run into errors on placing values it's due to you already having duplicate entries in your db, I was safely able to just delete the duplicates (since the data is in the component_type null entries already). If you rerun a fresh import this can be avoided.
7) repeat through all "_component" tables
8) Backup your db once you've done them all so you have a checkpoint.

**Comment by mitenka** [comment]
Thank you @Kyle772. Here's the SQL query (thank to ChatGPT) I used to find all corrupted records in all tables which names end with "_components":
```
DO $$
DECLARE
    v_table_name TEXT;
    v_column_name TEXT := 'component_type';
    v_record_count INTEGER;
BEGIN
    FOR v_table_name IN
        SELECT table_name
        FROM information_schema.tables
        WHERE table_schema = 'public'
          AND table_name LIKE '%_components'
          AND table_type = 'BASE TABLE'
    LOOP
        EXECUTE format('
            SELECT COUNT(*) FROM %I WHERE %I IS NULL
        ', v_table_name, v_column_name) INTO v_record_count;
        IF v_record_count > 0 THEN
            RAISE NOTICE 'Table % contains % records where % is NULL', v_table_name, v_record_count, v_column_name;
        END IF;
    END LOOP;
END $$;
```

**Comment by Baboo7** [comment]
@Kyle772 @mitenka @exploevo @ZiggyReQurv this bug is fixed in version 1.19.0. Thank you @mitenka, your PR helped me a lot 🙏 😌
